packages/qdrant-loader-mcp-server/src/qdrant_loader_mcp_server/search/components/vector_search_service.py

```
@@ -124,6 +144,74 @@ def _cleanup_expired_cache(self) -> None:
+        """
+        if self._collection_capabilities is not None:
+            return self._collection_capabilities
+
+        async with self._capabilities_lock:
+            if self._collection_capabilities is not None:
+                return self._collection_capabilities
+            try:
+                info = await self.qdrant_client.get_collection(
+                    collection_name=self.collection_name
+                )
+            except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)